### PR TITLE
add user docs for ssr assist

### PR DESCRIPTION
**Pull request by JoshMcguigan**
@matklad 

This is a small follow up on #7874, adding user docs for the SSR assist functionality. Since most other assists aren't handled this way I wasn't sure exactly how we wanted to document this, so feel free to suggest alternatives.

### CODE REVIEW

File: crates/ide_ssr/src/lib.rs [hunk 1]
```diff
@@ -57,6 +57,15 @@
 //
 // | VS Code | **Rust Analyzer: Structural Search Replace**
 // |===
+//
+// Also available as an assist, by writing a comment containing the structural
+// search and replace rule. You will only see the assist if the comment can
+// be parsed as a valid structural search and replace rule.
+//
+// ```rust
+// // Place the cursor on the line below to see the assist.
```
Comment (matklad) [inline crates/ide_ssr/src/lib.rs:66]:
```suggestion
// // Place the cursor on the line below to see the assist 💡.
```

**Comment by matklad** [comment]
lgtm!

bors d+

File: crates/ide_ssr/src/lib.rs (hunk 1 above)
Comment (JoshMcguigan) [reply crates/ide_ssr/src/lib.rs]: :+1: good suggestion!

**Pull request merged**